#1095 Fix crash for shares larger than 2GB. #1089 add ETA/bytes uploaded in progress bar of ReceiveMode items

mig5 opened the pull request:
Fixes #1089 
Fixes #1095 

The progress bar crash in #1095 has something to do with 32 bit integers in C/Python (I don't quite understand it) - a 2GB file, expressed in bytes, exceeded that integer and caused the crash. 

I copied the fix from the other open source project mentioned in that ticket, and it seems to work.

See also https://www.networkworld.com/article/3010974/whats-so-special-about-2147483648.html

SaptakS commented on onionshare_gui/tab/mode/history.py:
```diff
@@ -461,7 +463,7 @@ def update(self, downloaded_bytes):
         self.downloaded_bytes = downloaded_bytes
 
         self.progress_bar.setValue(downloaded_bytes)
```
Shouldn't this line also be `self.progress_bar.setValue(downloaded_bytes)` ?

mig5 commented:
I *think* I have fixed it, but if anything it's almost accidental, as I feel a bit muddled between all the total_bytes vs downloaded_bytes and which ones need to be converted to MB or why others don't seem to.

So I'd appreciate others closely looking at this code (not just the bits I changed but those around it). Maybe there's a more DRY way than all these `/ 1024` calculations.
